helpdesk_mgmt/models/helpdesk_ticket.py

```diff
@@ -100,7 +100,7 @@ def _name_search(
         if name:
             domain = ['|', ('number', operator, name), ('name', operator, name)]
             if operator in expression.NEGATIVE_TERM_OPERATORS:
-                domain = ['&', '!'] + domain[1:]
+                domain = domain[1:]
```

Comment:
I copied `domain = ['&', '!'] + domain[1:]` from `_name_search` of `account.account`.
Do you think there is also a problem there?

Reply:
You can check in the included test that it seems to work. For me also makes sense, because e.g. if operator is "equals", we have

("number" equals name) OR ("name" equals name)

, but if operator is "not equals" we should need

("number" not equals name) AND ("name" not equals name)

in order to achieve the desired effect. I don't know if account.account works properly, maybe there's a bug there, maybe I'm wrong... So for that I made the test

Then, other equivalent code should be (even better for further domain additions)

domain[0] = "&"